src/services/transaction_watcher_signer_service.ts

```diff
@@ -345,9 +404,21 @@ export class TransactionWatcherSignerService {
                 }
                 if (tx.blockNumber + NUMBER_OF_BLOCKS_UNTIL_CONFIRMED < latestBlockNumber) {
                     tx.status = TransactionStates.Confirmed;
+                    this._transactionsUpdateCounter.inc({ signer_address: tx.from, status: tx.status }, 1);
                     await this._transactionRepository.save(tx);
                 }
             }
         }
     }
+    private async _updateSignerBalancesAsync(): Promise<void> {
+        for (const signerAddress of this._availableSignerPublicAddresses) {
+            // TODO(oskar) - use contract to grab all balances in a single RPC call?
+            const signerBalance = await this._web3Wrapper.getBalanceInWeiAsync(signerAddress);
```

Comment:
No harm, although the only reason this could fail is the RPC call, so there is a chance that all of them will fail and will have to wait for the next metric update interval tick with the hope that the RPC comes back.